Extension/test/unitTests/updowngrade.test.ts

```diff
@@ -0,0 +1,141 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from "assert";
+import { resolveVariables, escapeForSquiggles } from "../../src/common";
+import { Build, Asset, getTargetBuild } from "../../src/githubAPI";
+import { PackageVersion } from '../../src/packageVersion';
+
+
+suite("UpgradeDowngrade", () => {
+
+    const updateChannel: string = "Insiders";
+
+    const asset_win32: Asset = {name: "cpptools-win32.vsix", browser_download_url: "https://github.com/microsoft/vscode-cpptools/releases/download/0.27.0/cpptools-win32.vsix"};
+    const asset_linux: Asset = {name: "cpptools-linux.vsix", browser_download_url: "https://github.com/microsoft/vscode-cpptools/releases/download/0.27.0/cpptools-linux.vsix"};
+    const asset_linux32: Asset = {name: "cpptools-linux32.vsix", browser_download_url: "https://github.com/microsoft/vscode-cpptools/releases/download/0.27.0/cpptools-linux32.vsix"};
+    const asset_osx: Asset = {name: "cpptools-osx.vsix", browser_download_url: "https://github.com/microsoft/vscode-cpptools/releases/download/0.27.0/cpptools-osx.vsix"};
+    const four_assets: Asset[] = [asset_win32, asset_linux, asset_linux32, asset_osx];
+
+    const release1: string = "0.27.1";
+    const insider3: string = "0.27.1-insiders3";
+    const insider2: string = "0.27.1-insiders2";
+    const insider1: string = "0.27.1-insiders";
+    const release0: string = "0.27.0";
+
+    suite("Downgrade", () => {
+        suite("Internal Testing, no Downgrade", () => {
+            test("Insider to Release", () => {
+                const builds: Build[] = [{
+                    name: release0, assets: four_assets}];
+
+                const userVersion: PackageVersion = new PackageVersion(insider1);
+                const targetBuild: Build | undefined = getTargetBuild(builds, userVersion, updateChannel);
+                assert.equal(targetBuild, undefined);
+            });
+
+            test("Insider to Insider", () => {
+                const builds: Build[] = [{
+                    name: insider2, assets: four_assets}, {
+                    name: insider1, assets: four_assets}, {
+                    name: release0, assets: four_assets}];
+
+                const userVersion: PackageVersion = new PackageVersion(insider3);
+                const targetBuild: Build | undefined = getTargetBuild(builds, userVersion, updateChannel);
+                assert.equal(targetBuild, undefined);
+            });
+        });
+        suite("Insider Users, Downgrade", () => {
```

Comment:
Can you also add a test that downgrades past more than one recalled release? (e.g. 0.27.1-insiders2 -> 0.27.0 or 0.27.1-insiders3 -> 0.27.1-insiders)